assets/source/setup-guide/app/steps/SetupTracking.js

```diff
@@ -218,9 +225,66 @@ const SetupTracking = ( { view } ) => {
 
 			setIsSaving( false );
 		},
-		[ appSettings, setIsSaving, setAppSettings, createNotice ]
+		[
+			appSettings,
+			setIsSaving,
+			setAppSettings,
+			connectAdvertiser,
+			createNotice,
+		]
 	);
 
+	const connectAdvertiser = useCallback( async () => {
```

Comment:
I suggest making this `connectAdvertiser` accept the advertiser ID and tracking tag as parameters, so it doesn't need to depend on `appSettings`, and to make it clearer that the action of connecting advertiser requires the two parameters.